Supplementary_data/notebookapp_imageexport.py

```diff
@@ -0,0 +1,305 @@
+        query_params = {
+            'time': (str(start_date), str(end_date)),
+            'geopolygon': geopolygon
+        }
+
+        # Find matching datasets
+        dss = [
+            dc.find_datasets(product=i, **query_params)
+            for i in sat_params[satellites]['products']
+        ]
+        dss = list(itertools.chain.from_iterable(dss))
+
+        # Get CRS and sensor
+        crs = str(dss[0].crs)
+        sensor = dss[0].metadata_doc['properties']['eo:platform'].capitalize()
+        sensor = sensor[0:-1].replace('_', '-') + sensor[-1].capitalize()
+
+        # Meets pansharpening requirements
+        can_pansharpen = style == 'True colour' and not resolution and sensor in [
+            'Landsat-7', 'Landsat-8'
+        ]
+
+        # Set up load params
+        if pansharpen and can_pansharpen:
+            load_params = {
+                'output_crs': crs,
+                'resolution': (-15, 15),
+                'align': (7.5, 7.5),
+                'resampling': 'bilinear'
+            }
+
+            # Add panchromatic to list of true colour bands
+            sat_params[satellites]['styles']['True colour'] += [
+                'nbart_panchromatic'
+            ]
+
+        else:
+
+            # Use resolution if provided, otherwise use default
+            if resolution:
+                sat_params[satellites]['resolution'] = resolution
+
+            load_params = {
+                'output_crs': crs,
+                'resolution': sat_params[satellites]['resolution'],
+                'resampling': 'bilinear'
+            }
+
+        # Load data from datasets
+        ds = dc.load(datasets=dss,
+                     measurements=sat_params[satellites]['styles'][style],
+                     group_by='solar_day',
+                     dask_chunks={
+                         'time': 1,
+                         'x': 3000,
+                         'y': 3000
+                     },
+                     **load_params,
+                     **query_params)
+        ds = masking.mask_invalid_data(ds)
+
+        # Create plain numpy array, optionally after pansharpening
+        if pansharpen and can_pansharpen:
+
+            # Perform Brovey pan-sharpening and return three numpy.arrays
+            print(f'Pansharpening {sensor} image to 15 m resolution')
+            red_sharpen, green_sharpen, blue_sharpen = pan_sharpen_brovey(
+                band_1=ds.nbart_red,
+                band_2=ds.nbart_green,
+                band_3=ds.nbart_blue,
+                pan_band=ds.nbart_panchromatic)
+            rgb_array = np.vstack([red_sharpen, green_sharpen, blue_sharpen])
+
+        else:
+            rgb_array = ds.isel(time=0).to_array().values
+
+        ############
+        # Plotting #
+        ############
+
+        # Create unique file name
+        site = reverse_geocode(coords=centre_coords)
+        fname = f"{sensor} - {date} - {site} - {style}, {load_params['resolution'][1]} m resolution.png"
```

Comment:
Hey @robbibt ! can we please replace the spaces and commas in this filename template with underscores? e.g. f"{sensor} _{date}_{site}_{style}_{load_params['resolution'][1]}_m_resolution.png" . Site might also need string processing to replace any special characters e.g. " ',+,& " .

Reply:
Hey @BexDunn! The current filename format is based on the DEA Images and Animations library naming convention here, so that visualisations produced by the tool can be easily added to those folders: https://geoscienceau.sharepoint.com/:w:/s/DEA/EWBOrEQPfnJOgk8hNveqFo0BT6KaoLbiuWIgiEolx_n29g?e=4MBZbf

It's been used there for a while and if we were to change it we'd need to go and re-name all the files in those folders as well... I do see the value in having a machine-readable option as well, so what if I added an option to the tool to set human-readable or machine readable file formats?